File: 3sum-closest.py
# http://oj.leetcode.com/problems/3sum-closest/
class Solution:
	# @return an integer
	def threeSumClosest(self, num, target):
		if len(num) < 3:
			return 0
		mn = abs(target - (num[0] + num[1] + num[2]))
		result = num[0] + num[1] + num[2]
		for i in range (0, len(num) - 2):
			for j in range (i+1, len(num) - 1):
				for k in range (j+1, len(num)):
					if abs(target - (num[i] + num[j] + num[k])) < mn:
						mn = abs(target - (num[i] + num[j] + num[k]))
						result = num[i] + num[j] + num[k]
						if mn == 0:
							return result
		return result
s = Solution()
a = [-1, 2, 1, -4]
print(s.threeSumClosest(a, 1))

File: 3sum-closest.cpp
// http://oj.leetcode.com/discuss/2217/o-n-3-solution-can-pass-the-judge 
// Python TLE
class Solution {
public:

    int threeSumClosest(vector<int> &num, int target) {

        int i,j,k,s,s1;

        s = num[0] + num[1] + num[2];
        s1 = abs(s - target);
        for(i=0;i<num.size()-2;i++)
            for(j=i+1;j<num.size()-1;j++)
                for(k=j+1;k<num.size();k++)
                {
                    if(s1> abs(num[i] + num[j] + num[k] - target) )
                    {
                        s = num[i] + num[j] + num[k];
                        s1 = abs(s - target);

                        // Without this condition, I get TLE
                        if(s1==0)
                            return s;
                    }
                }
        return s;
    }
};

